test/nightly/run_combined_tests.py

```diff
@@ -0,0 +1,464 @@
+        return False, error_msg
+
+
+def run_accuracy_for_model(
+    model: ModelLaunchSettings,
+    base_url: str,
+    eval_name: str,
+    num_examples: Optional[int],
+    num_threads: int,
+    max_tokens: Optional[int] = None,
+    return_latency: bool = False,
+    **eval_kwargs,
+) -> Tuple[bool, Optional[str], Optional[dict]]:
+    """Run accuracy test for a single model using run_eval (SimpleNamespace).
+
+    Args:
+        model: ModelLaunchSettings with model config
+        base_url: Server base URL
+        eval_name: Evaluation name (e.g., "mgsm_en", "mmmu", "gpqa")
+        num_examples: Number of examples (None means all)
+        num_threads: Number of threads for evaluation
+        max_tokens: Max tokens for generation
+        return_latency: Whether to return latency metrics
+        **eval_kwargs: Additional kwargs to pass to run_eval
+
+    Returns:
+        Tuple of (success, error_message, metrics_dict)
+    """
+    print(f"\n{'='*60}")
+    print(f"Running ACCURACY test for {model.model_path}")
+    print(f"  Eval: {eval_name}, Examples: {num_examples}")
+    print(f"{'='*60}\n")
+
+    process = None
+    try:
+        # Launch server
+        process = popen_launch_server(
+            model=model.model_path,
+            base_url=base_url,
+            other_args=model.extra_args,
+            timeout=DEFAULT_TIMEOUT_FOR_SERVER_LAUNCH,
+        )
+
+        # Build eval args
+        args = SimpleNamespace(
+            base_url=base_url,
+            model=model.model_path,
+            eval_name=eval_name,
+            num_examples=num_examples,
+            num_threads=num_threads,
+        )
+
+        # Add optional parameters
+        if max_tokens is not None:
+            args.max_tokens = max_tokens
+
+        if return_latency:
+            args.return_latency = True
+
+        # Add any extra kwargs
+        for key, value in eval_kwargs.items():
+            setattr(args, key, value)
+
+        # Run evaluation
+        result = run_eval(args)
+
+        # Handle result format (run_eval can return metrics or (metrics, latency))
+        if return_latency and isinstance(result, tuple):
+            metrics, latency = result
+            metrics["latency"] = round(latency, 4)
+        else:
+            metrics = result
+
+        print(f"✓ Accuracy test succeeded for {model.model_path}: {metrics}")
+        return True, None, metrics
+
+    except Exception as e:
+        error_msg = f"Accuracy test exception for {model.model_path}: {str(e)}"
+        print(f"✗ {error_msg}")
+        return False, error_msg, None
+
+    finally:
+        if process:
+            kill_process_tree(process.pid)
+
+
+def run_metrics(
+    models: List[ModelLaunchSettings],
+    run_perf: bool = True,
+    run_accuracy: bool = True,
+    is_vlm: bool = False,
+    base_url: Optional[str] = None,
+    profile_dir: Optional[str] = None,
+    test_name: str = "NightlyMetrics",
+    # Performance test parameters
```

Comment:
Can we put arguments for accuracy and performance test parameters into separate classes? E.g. 

class AccuracyTestParams: (dataset_name, num_examples, baseline_accuracy)

class PerformanceTestParams: (batch_size, ITL, OTL, etc.)